ext/hal/nordic/nrfx_config_nrf52840.h

```diff
@@ -0,0 +1,2974 @@
+#ifndef NRFX_CONFIG_NRF52840_H__
+#define NRFX_CONFIG_NRF52840_H__
+
+// <<< Use Configuration Wizard in Context Menu >>>\n
+
+// <h> nRF_Drivers
+
+// <e> NRFX_CLOCK_ENABLED - nrfx_clock - CLOCK peripheral driver
+//==========================================================
+#ifndef NRFX_CLOCK_ENABLED
+#define NRFX_CLOCK_ENABLED 1
+#endif
+// <o> NRFX_CLOCK_CONFIG_LF_SRC  - LF Clock Source
+
+// <0=> RC
+// <1=> XTAL
+// <2=> Synth
+
+#ifndef NRFX_CLOCK_CONFIG_LF_SRC
+#define NRFX_CLOCK_CONFIG_LF_SRC 1
```

Comment:
How does this configuration system integrate with Kconfig?

If it was documented to users what this file was, what would that documentation say?

Reply:
This file provides static configuration for nrfx drivers. This way you can tailor them to your specific needs: exclude some features or support for particular instances of peripherals, specify default values for configuration structures, etc.
Regarding integration with Kconfig - CONFIG_* macros can be used in this file if there's a need to influence the compilation of nrfx drivers. Changes to this file will come later, together with shims adapting the drivers to Zephyr's APIs.

Reply:
Right, so in Zephyr that use-case is covered by Kconfig. Users should not need to learn a new mechanism to be able to configure a new subsystem in Zephyr. This does not have to block the merge, but eventually we will need to have the ability to configure every single on of these configs through Kconfig.

I suspect you will be able to extend your nrf_config_*.h generator to also be able to generate a Kconfig file. After you have that in place it might be possible to have a simple adapter that does

#define NRFX_CLOCK_CONFIG_LF_SRC CONFIG_NRFX_CLOCK_CONFIG_LF_SRC

for each config.